modules/runner/src/main/ts/core/Runnable.ts

```diff
@@ -6,6 +6,15 @@ export const createRunnable = (title: string, fn: ExecuteFn | undefined): Runnab
   let timeout = -1;
   let slow = 75;
   let state = RunnableState.NotRun;
+  const listeners: Record<string, Set<(value: number) => void>> = {};
+
+  const fireChange = (type: string, value: number) => {
```

Comment:
Should type be `retries | slow | timeout` as per the `_onChange` method? And should that type be pulled out

Reply:
It should not, as I intended for this part to be generic/work for anything largely. We only need the types for the public API on what can be registered as listeners to make sure someone doesn't accidentally listen on the wrong thing.